app/ide-desktop/lib/client/src/bin/server.ts

```diff
@@ -98,10 +121,20 @@ export class Server {
     /** Server constructor. */
     static async create(config: Config): Promise<Server> {
         const localConfig = Object.assign({}, config)
-        localConfig.port = await findPort(localConfig.port)
-        const server = new Server(localConfig)
-        await server.run()
-        return server
+
+        if (await isPortInUse(localConfig.port)) {
+            logger.error(
+                `💥Port ${localConfig.port} is already in use. Please close the application using it and try again.`
+            )
+
+            return Promise.reject(
+                `💥Port ${localConfig.port} is already in use. Please close the application using it and try again.`
+            )
+        } else {
+            const server = new Server(localConfig)
```

Comment:
I mentioned this approach here: https://github.com/enso-org/enso/pull/9965#discussion_r1601638151

Lemme think about how we can tackle this...

Reply:
If we start the server, we're getting this error:
```
(node:65794) electron: Failed to load URL: https://localhost:8080/?engine.projectManagerUrl=ws%3A%2F%2F127.0.0.1%3A30535 with error: ERR_SSL_PROTOCOL_ERROR
```
which is, IMO absolutely uninformative and the server notified that it successfully started on the port and it is weird.

Reply:
As an alternative, I can put the check later to reduce the possible time lag between checking the port and starting the server. 

Also, I think even if the logic was all sync, it can't be reliable at 100% (please let me know if I'm wrong).

On the other though, if we try to launch 2 IDE on the same machine we'd also fail in production...

Reply:
Will raise this question tommorow on daily, will see, maybe we'll rewrite this part completely as it also has security issues.

Reply:
We decided to roll back the logic back to the original (start on free next port).